File: Yow.CoD.Finance.Domain/Services/CreateLoanCommandHandler.cs
using System.Threading.Tasks;
using Yow.CoD.Finance.Domain.Contracts;
using Yow.CoD.Finance.Domain.Model;

namespace Yow.CoD.Finance.Domain.Services
{
    public class CreateLoanCommandHandler : IHandler<CreateLoanCommand>
    {
        private readonly IRepository<Loan> _repository;

        public CreateLoanCommandHandler(IRepository<Loan> repository)
        {
            _repository = repository;
        }

        public async Task Handle(CreateLoanCommand command)
        {
            var loan = new Loan(command.LoanId);
            loan.Create(command);
            await _repository.Save(loan);
        }
    }
}

File: java/domain/src/main/java/com/leecampbell/cod/domain/services/CreateLoanCommandHandler.java
package com.leecampbell.cod.domain.services;

import com.leecampbell.cod.domain.contracts.*;
import com.leecampbell.cod.domain.model.Loan;

public final class CreateLoanCommandHandler implements CommandHandler<CreateLoanCommand, Receipt> {
    private final Repository _repository;

    public CreateLoanCommandHandler(Repository repository) {
        _repository = repository;
    }

    public Receipt handle(CreateLoanCommand command) {
        Loan loan = _repository.get(command.getAggregateId());
        loan.create(command);
        _repository.save(loan);
        return new CommandReceipt(loan.getId(), loan.getVersion());
    }
}